pkg/trace/agent/run.go

```diff
@@ -124,8 +127,20 @@ func Run(ctx context.Context) {
 
 	rand.Seed(time.Now().UTC().UnixNano())
 
+	var t tagger.Tagger
+	if coreconfig.Datadog.GetBool("apm_config.remote_tagger") {
+		t = remote.NewTagger()
+	} else {
+		t = local.NewTagger(collectors.DefaultCatalog)
+	}
+	tagger.SetDefaultTagger(t)
```

Comment:
Why are we overriding the default tagger if the boolean is not set? 
If we ever change the default, then we won't see the change propagate as we might expect because we are overriding it here.

Default:
https://github.com/DataDog/datadog-agent/blob/b74f855a193d077a2e47f78a6dacde5ba03d3012/pkg/tagger/global.go#L129-L131

Reply:
I want to eventually get rid of the default entirely. It's not possible right now because it breaks a lot of tests, but a cleanup is incoming in the next release cycle.